packages/grafana-i18n/src/i18n.tsx

```diff
@@ -0,0 +1,155 @@
+import i18n, { InitOptions, TFunction } from 'i18next';
+import LanguageDetector, { DetectorOptions } from 'i18next-browser-languagedetector';
+import { ReactElement } from 'react';
+// eslint-disable-next-line no-restricted-imports
+import { initReactI18next, Trans as I18NextTrans } from 'react-i18next';
+
+import { DEFAULT_LANGUAGE, PSEUDO_LOCALE } from './constants';
+import { LANGUAGES } from './languages';
+import { TransProps, TransType } from './types';
+
+let tFunc: TFunction<string[], undefined> | undefined;
+let transComponent: TransType;
+
+export async function initPluginTranslations(id: string) {
+  // If the resources are not an object, we need to initialize the plugin translations
+  if (!getI18nInstance().options?.resources || typeof getI18nInstance().options.resources !== 'object') {
+    await getI18nInstance().use(initReactI18next).init({
+      resources: {},
+      returnEmptyString: false,
+      lng: DEFAULT_LANGUAGE, // this should be the locale of the phrases in our source JSX
+    });
+  } else {
+    // passes i18n down to react-i18next
+    getI18nInstance().use(initReactI18next).init(getI18nInstance().options);
```

Comment:
This would be interesting to remove, but that would mean we need to externalize `react-i18next` to and backport that change

Reply:
Fixed 👍